src/java.base/share/classes/jdk/internal/foreign/abi/ppc64/aix/AixPPC64VaList.java

```diff
@@ -33,123 +33,247 @@
+			}
+		}
+
+		return argLayoutSize;
+	}
+
+	/* Check whether the argument to be skipped exceeds the existing memory size in the VaList */
+	private void checkNextArgument(MemoryLayout argLayout, long argByteSize) {
+		if (argByteSize > segment.byteSize()) {
+			throw SharedUtils.newVaListNSEE(argLayout);
+		}
+	}
+
+	@Override
+	public void skip(MemoryLayout... layouts) {
+		Objects.requireNonNull(layouts);
+		sessionImpl().checkValidState();
+
+		for (MemoryLayout layout : layouts) {
+			Objects.requireNonNull(layout);
+			long argByteSize = getAlignedArgSize(layout);
+			checkNextArgument(layout, argByteSize);
+			/* Skip to the next argument in the va_list buffer */
+			segment = segment.asSlice(argByteSize);
+		}
+	}
+
+	public static VaList ofAddress(MemoryAddress addr, MemorySession session) {
+		MemorySegment segment = MemorySegment.ofAddress(addr, Long.MAX_VALUE, session);
+		return new AixPPC64VaList(segment, session);
+	}
+
+	@Override
+	public MemorySession session() {
+		return session;
+	}
+
+	@Override
+	public VaList copy() {
+		sessionImpl().checkValidState();
+		return new AixPPC64VaList(segment, session);
+	}
+
+	@Override
+	public MemoryAddress address() {
+		return segment.address();
+	}
+
+	@Override
+	public String toString() {
+		return "AixPPC64VaList{" + segment.address() + '}';
+	}
+
+	static Builder builder(MemorySession session) {
+		return new Builder(session);
+	}
+
+	public static non-sealed class Builder implements VaList.Builder {
+		private final MemorySession session;
+		private final List<SimpleVaArg> stackArgs = new ArrayList<>();
+
+		public Builder(MemorySession session) {
+			MemorySessionImpl.toSessionImpl(session).checkValidState();
+			this.session = session;
+		}
+
+		private Builder setArg(MemoryLayout layout, Object value) {
+			Objects.requireNonNull(layout);
+			Objects.requireNonNull(value);
+			stackArgs.add(new SimpleVaArg(layout, value));
+			return this;
+		}
+
+		@Override
+		public Builder addVarg(ValueLayout.OfInt layout, int value) {
+			return setArg(layout, value);
+		}
+
+		@Override
+		public Builder addVarg(ValueLayout.OfLong layout, long value) {
+			return setArg(layout, value);
+		}
+
+		@Override
+		public Builder addVarg(ValueLayout.OfDouble layout, double value) {
+			return setArg(layout, value);
+		}
+
+		@Override
+		public Builder addVarg(ValueLayout.OfAddress layout, Addressable value) {
+			return setArg(layout, value.address());
+		}
+
+		@Override
+		public Builder addVarg(GroupLayout layout, MemorySegment value) {
+			return setArg(layout, value);
+		}
+
+		public VaList build() {
+			if (stackArgs.isEmpty()) {
+				return EMPTY;
+			}
+
+			/* All primitves/pointer (aligned with 8 bytes) are directly stored in the va_list buffer
+			 * and all elements of stuct are totally copied to the va_list buffer (instead of storing
+			 * the va_list address), in which  case we need to calculate the total byte size of the
```

Comment:
There is an extra space between "which" and "case".

Reply:
Fixed.